pkg/apis/serving/v1beta1/configuration_validation.go

```diff
@@ -61,3 +65,34 @@ func (cs *ConfigurationStatus) Validate(ctx context.Context) *apis.FieldError {
 func (csf *ConfigurationStatusFields) Validate(ctx context.Context) *apis.FieldError {
 	return nil
 }
+
+// ValidateLabels function validates service labels
+func (c *Configuration) ValidateLabels() (errs *apis.FieldError) {
+	for key, val := range c.GetLabels() {
+		switch {
+		case key == config.VisibilityLabelKey:
+			errs = errs.Also(validateClusterVisbilityLabel(val))
+		case key == serving.RouteLabelKey:
+		case key == serving.ServiceLabelKey:
+			errs = errs.Also(verifyLabelOwnerRef(val, serving.ServiceLabelKey, "Service", c.GetOwnerReferences()))
+		case strings.HasPrefix(key, serving.GroupNamePrefix):
+			errs = errs.Also(apis.ErrInvalidKeyName(key, ""))
+		}
+	}
+	return
+}
+
+// verifyServiceLabelOwnerRef function verifies the owner references of resource with label key has val value.
+func verifyLabelOwnerRef(val, label, resource string, ownerRefs []metav1.OwnerReference) (errs *apis.FieldError) {
+	found := false
+	for i, ref := range ownerRefs {
+		if ref.Kind != resource || val != ref.Name {
+			errs = errs.Also(apis.ErrInvalidArrayValue(ref.Name, label, i))
```

Comment:
I am not sure this is something we want to add to our errors here. Seems like a strange side-effect of `verifyLabelOwnerRef`. If I understand correctly this is ensuring that only the service has an ownership reference on this resource. So long as the service is an owner and the value is matches our value we should be good.

Reply:
What happens in inverse case where configuration resource has Service label but no owner reference, then is this acceptable or not?
The loop guards against this use case.

Reply:
True. So I think the logic here should validate:

If there is a label there should be a matching owner.

If there are extra owners that shouldn't be a problem for this function so long as one of them matches the resource type and name. With that you should be able to get rid of line 93, return when you find the matching owner, otherwise error if you exit without finding the owner.

Reply:
@vagababov raised concern on the [original PR](https://github.com/knative/serving/pull/4822#discussion_r305460508) to address the case of multiple service owners with mismatched keys.  

configuration resource with label
`service : a`

case 1
owner reference
service: a (valid)

case 2
owner reference
service: b (invalid)

case 3
service: c (ignored, no error)
service: a (valid)

case 4
no owner reference
(invalid)

case 5
random owner reference
(invalid)

@vagababov suggested adding an error in case 3 for service with mismatched names(if I understood your comment correctly) but @dgerd  you are suggesting ignoring the error as long as there exists 1 matching Service with correct name. 

Is this change okay with @dgerd @vagababov ? I am okay with this.

Reply:
Yes. Case 3 should never happen and if it does this does not seem like the proper place to enforce such a thing.

The cases laid out here all look good to me.

Reply:
Awesome. Thanks for quick reply. I have updated the PR to address the above cases.